assembly/src/ast/instruction/mod.rs

```diff
@@ -204,7 +205,7 @@ pub enum Instruction {
     CDropW,
 
     // ----- input / output operations -----------------------------------------------------------
-    Push(ImmFelt),
+    Push(Immediate<HexEncodedValue>),
```

Comment:
I think we should use this opportunity to rename `HexEncodedValue` to `IntValue`, since it is used now in more places than just the handling of hex-encoded integers.